Makefile

```diff
@@ -0,0 +1,55 @@
+# if provided, FILTER is used as the --filter argument to `swift test`.
+ifdef FILTER
+	FILTERARG = --filter $(FILTER)
+else
+	FILTERARG =
+endif
+
+# if no value provided assume sourcery is in the user's PATH
+SOURCERY ?= sourcery
+
+define check_for_gem
+	gem list $(1) -i > /dev/null || gem install $(1) || { echo "ERROR: Failed to locate or install the ruby gem $(1); please install yourself with 'gem install $(1)' (you may need to use sudo)"; exit 1; }
+endef
+
+all:
+	swift build
+
+# project generates the .xcodeproj, and then modifies it to add
+# spec .json files to the project
+project:
+	swift package generate-xcodeproj
+	@$(call check_for_gem,xcodeproj)
+	ruby etc/add_json_files.rb
+
+linuxmain:
+	$(SOURCERY) --sources Tests/ --exclude-sources Tests/DisabledTests/ --templates Tests/LinuxMain.stencil --output Tests/LinuxMain.swift
+	@tail -n +5 Tests/LinuxMain.swift > Tests/LinuxMain.swift.tmp
+	@mv Tests/LinuxMain.swift.tmp Tests/LinuxMain.swift
+
+test:
+	swift test -v $(FILTERARG)
+
+test-pretty:
+	@$(call check_for_gem,xcpretty)
+	@set -o pipefail && swift test $(FILTERARG) 2>&1 | xcpretty
+
+lint:
+	@swiftformat .
```

Comment:
do we wnat to suppress all of the linting output with the `@`s (if I remember correctly what that does)? seems like it would be helpful to see what went wrong printed out? (same re test-pretty)

Reply:
It suppresses printing out the command not the output, I think make dumps the full command by default cus its usually a big constructed gcc call or w.e. I figured here there's nothing being constructed, cleans up your output a bit, its entirely opinionated and loosely held

Reply:
oh right, yes that seems fine to me then!